### PR TITLE
Webyourmind/sept week 2

### LINKED ISSUE #108: Add eslint, eslint prettier plugin, travis CI
* Add eslint (see service repo for settings)
* Add eslint prettier plugin. This plugin incorporates prettier as part of linting. Regular contributors will most likely have prettier integrated into their editor on-save. But for new contributors or for those that want to run it by hand, they can view/fix violations with `eslint`/`eslint --fix`. This also will allow us to validate in PRs. (we need this in service as well)
* Run eslint as part of `npm test` (see service repo)
* Add travis config for PR build (see service repo)

### LINKED ISSUE #287: remove stale code
There are a number of routes and components left over from old UI approaches (e.g., Curations). Remove this.

### PR DESCRIPTION
**Pull request by storrisi**
Including:
- Fixes #108 
- Fixes #287

### CODE REVIEW

**Comment by storrisi** [comment]
Updated PR with fix #287

**Comment by storrisi** [comment]
PR can now be reviewed

File: src/utils/routingConstants.js [hunk 1]
```diff
@@ -4,8 +4,8 @@
 export const ROUTE_ABOUT = '/about'
 export const ROUTE_DEFINITIONS = '/definitions'
 export const ROUTE_SHARE = '/share'
-export const ROUTE_INSPECT = '/inspect'
-export const ROUTE_CURATE = '/curate'
+// export const ROUTE_INSPECT = '/inspect'
```
Comment (jeffmcaffer) [inline src/utils/routingConstants.js]: go ahead and remove

File: src/components/FilterBar.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ export default class FilterBar extends Component {
     const { options, defaultValue, value, onSearch } = this.props
     return (
       <AsyncTypeahead
-        ref="typeahead"
+        ref={ref => (this.typeahead = ref)}
```
[the remark this answers is not in the document]
Comment (gianpaj) [reply src/components/FilterBar.js]: Good catch on the `ref`. I must have confused the React docs.

Regarding the `.open()`, I believe this is for the modals e.g. the `ContributePrompt`. Is that right?
The state of the modal (showing or not) is inside the `ContributePrompt` component. Should I move it outside and add a prop to be able to invoke a close method from inside the component?

**Comment by gianpaj** [comment]
Hi @jeffmcaffer, I've made the changes as requested as well as removed 2 pages which I didn't before.

One remaining point is regarding the `.open()` modal approach. Let me know if I should do it as I suggested here: https://github.com/clearlydefined/website/pull/300#discussion_r218006355

Thanks

File: src/components/FilterBar.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (AlexWebYourmind) [reply src/components/FilterBar.js]: @storrisi @jeffmcaffer any updates on this?

**Pull request merged**